app/services/category_service.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+# CategoryService
+class CategoryService
+  include ApplicationHelper
+  attr_reader :category, :page_number, :filter, :books, :active_more_btn
+
+  def initialize
+    @active_more_btn = false
+  end
+
+  def call(params)
+    @page_number = 1
```

Comment:
Assigning instance variables should be in the controller. No need for service. Move this code to the controller.
If you have a complex logic of finding and filtering books this logic can be moved to the query object: https://www.sitepoint.com/7-design-patterns-to-refactor-mvc-components-in-rails/